enhancements/cluster-logging/elasticsearch-v2-api-and-infrastructure.md

```diff
@@ -0,0 +1,319 @@
+    * This includes breaking up nodes by roles
+    * Specifying (usually pass through) of node requirements
+
+With this proposal the workflow would be:
+
+1. Create their elasticsearch CR specifying a particular sized cluster (based on t-shirt sizes)
+
+Current specification:
+
+```yaml
+apiVersion: "logging.openshift.io/v1"
+kind: "Elasticsearch"
+metadata:
+  name: "elasticsearch"
+spec:
+  managementState: "Managed"
+  nodeSpec:
+    resources:
+      limits:
+        memory: 1Gi
+      requests:
+        cpu: 100m
+        memory: 1Gi
+  nodes:
+  - nodeCount: 1
+    roles:
+    - client
+    - data
+    - master
+    storage:
+      storageClassName: gp2
+      size: 10Gi
+  redundancyPolicy: ZeroRedundancy
+  indexManagement:
+    policies:
+    - name: infra-policy
+      pollInterval: 1m
+      phases:
+        hot:
+          actions:
+            rollover:
+              maxAge:   2m
+        delete:
+          minAge: 5m
+    mappings:
+    - name:  infra
+      policyRef: infra-policy
+      aliases:
+      - infra
+      - logs.infra
+```
+
+Proposed:
+
+```yaml
+apiVersion: "logging.openshift.io/v1"
+kind: "Elasticsearch"
+metadata:
+  name: "elasticsearch"
+spec:
+  storage:
+    storageClassName: gp2
+    size: 10Gi
+  size: 0x.medium
+  indexManagement:
+    infraPolicies:
+    - phases:
+      hot:
+        actions:
+          rollover:
+            maxAge:   2m
+      delete:
+        minAge: 5m
+    appPolicies:
+    - phases:
+      hot:
+        actions:
+          rollover:
+            maxAge: 2m
+      delete:
+        minAge: 5m
+```
+
+## Proposal
+
+Currently the Elasticsearch CR is overly complicated and allows for too much granular control over the data nodes, which places the expectation
+of expertise on the user rather than containing it to the operator (which is where it should reside). This proposal seeks to outline a v2 to provide
+a more simple means of interacting with the Elasticsearch Operator to ask for a managed Elasticsearch cluster that can be better
+configured for performance.
+
+### User Stories
+
+#### As an user of EO, I want to have to provide minimal information for my cluster to be created for me
+
+As an user I want to just be able to provide information such as a t-shirt size (which is outlined further for me) and my storage information
+and have an elasticsearch cluster spun up for me so that I do not need to be an expert and configure my node roles and node resources.
+
+### Implementation Details
+
+#### Assumptions
+
+* Currently the EO CRD version is V1 and this would be a V2. We would assume that EO is able to react for both a V1 and V2 and if the user creates a V2
+  elasticsearch CR with the same name as an existing V1, the operator would migrate the V1 backing storage naming to be used for V2 (read: it would
+  rename the PVCs and rebind the backing PV so that it matches the updated naming convention).
```

Comment:
Is renaming a PVC possible? IIRC the only changes allowed on PVCs after creation are changes on metadata labels and annotations.

Reply:
you can't change the name of the pvc.. maybe i should reword this. the idea is we would change the name we use for PVC creation so we would create a PVC that follows that naming convention and then rebind the old PVC's backing PV to this new PVC and then remove the old one... effectively renaming.